doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -425,16 +425,98 @@ Displaying information about notations
      (corresponding to :token:`ltac_expr` in the documentation).
    - `vernac` - for :token:`command`\s
 
-   The first three of these give the precedence and associativity for each construct.
-   For example, these lines printed by `Print Grammar tactic` indicates that the `try` construct
-   is at level 3 and right-associative.  `SELF` represents the `tactic_expr` nonterminal
-   at level 5 (the top level)::
-
+   This command doesn't display all nonterminals of the grammar.  For example,
+   productions shown by `Print Grammar tactic` refer to nonterminals `tactic_then_locality`
+   and `tactic_then_gen` which are not shown and can't be printed.
```

Comment:
I have a branch which add supports for `Print Grammar` of any entry. I'm not fully satisfied by it, though because collisions can happen, and entries should probably structured. E.g. an API with commands such as `Print Grammar Tactic tactic_then_gen` or `Print Grammar Constr fields_def` would already be better. Maybe would we need even stronger structuration.

Reply:
We'd need to figure out how to handle locally-defined nonterminals that have the same name.  How would we show such nonterminals to users in a non-confusing way?  We could somehow require unique names or we'd need a way of qualifying name (ugh).  Ltac2 has 35 or more such symbols while the rest of the grammar has maybe 6 or 7.

I'm `Print Grammar` is hugely useful in the long term.  Once the grammar in the doc is updated and we update the mlgs to match that as much as possible, it's probably better for users to look in the doc.  Of course, the doc won't show productions added by notations, but these additions appear only on specific nonterminals.  The doc would not cover additions from plugins external to the Coq source tree either.

Reply:
> I have a branch which add supports for Print Grammar of any entry

I think it's a bad idea to export such internals to end-users. A better way to do that would be to have a way to recursively extract all entries accessible from a few selected roots and display that to the user, with on-the-fly deambiguation of the entry names.

Reply:
I'm not far from thinking that this is indeed too internal to be worth being exposed. However, it happens to be useful for debugging. Maybe should it be reserved to developers.